org.eclipse.scanning.example/src/org/eclipse/scanning/example/malcolm/DummyMalcolmRecord.java

```diff
@@ -0,0 +1,453 @@
+            PVStructure returnPvStructure = pvDataCreate.createPVStructure(mapStructure);
+            
+            if ("validate".equals(methodName)) {
+            	returnPvStructure = args;
+            } else if ("configure".equals(methodName)) {
+            	pvRecord.getPVStructure().getSubField(PVString.class, "state.value").put("CONFIGURING");
+            } else if ("run".equals(methodName)) {
+                pvRecord.getPVStructure().getSubField(PVString.class, "state.value").put("RUNNING");
+                try {
+    				Thread.sleep(2000);
+    			} catch (InterruptedException e1) {
+    				e1.printStackTrace();
+    			}
+            }
+
+
+        	pvRecord.getPVStructure().getSubField(PVString.class, "state.value").put("READY");
+            
+        	pvRecord.releaseControl();
+            callback.requestDone(statusOk, returnPvStructure);
+            return;
+            
+        }
+
+        private void handleError(String message, RPCResponseCallback callback, boolean haveControl)
+        {
+            if (haveControl)
+                pvRecord.releaseControl();
+            Status status = StatusFactory.getStatusCreate().
+                    createStatus(StatusType.ERROR, message, null);
+            callback.requestDone(status, null);
+        }
+    }
+
+    public static DummyMalcolmRecord create(String recordName)
+    {
+        FieldBuilder fb = fieldCreate.createFieldBuilder();
+        
+
+        Structure metaStructure = fieldCreate.createFieldBuilder().
+    			add("description", ScalarType.pvString).
+    			addArray("tags", ScalarType.pvString).
+    			add("writeable", ScalarType.pvBoolean).
+    			add("label", ScalarType.pvString).
+    			setId("malcolm:core/BlockMeta:1.0").
+    			createStructure();
+        
+        Structure choiceMetaStructure = fb.
+    			add("description", ScalarType.pvString).
+    			addArray("choices", ScalarType.pvString).
+    			addArray("tags", ScalarType.pvString).
+    			add("writeable", ScalarType.pvBoolean).
+    			add("label", ScalarType.pvString).
+    			setId("malcolm:core/ChoiceMeta:1.0").
+                createStructure();
+        
+        Structure stringMetaStructure = fieldCreate.createFieldBuilder().
+    			add("description", ScalarType.pvString).
+    			addArray("tags", ScalarType.pvString).
+    			add("writeable", ScalarType.pvBoolean).
+    			add("label", ScalarType.pvString).
+    			setId("malcolm:core/StringMeta:1.0").
+    			createStructure();
+        
+        Structure booleanMetaStructure = fieldCreate.createFieldBuilder().
+    			add("description", ScalarType.pvString).
+    			addArray("tags", ScalarType.pvString).
+    			add("writeable", ScalarType.pvBoolean).
+    			add("label", ScalarType.pvString).
+    			setId("malcolm:core/BooleanMeta:1.0").
+    			createStructure();
+        
+        Structure intNumberMetaStructure = fieldCreate.createFieldBuilder().
+    			add("dtype", ScalarType.pvString).

```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "dtype" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)